index.d.ts

```diff
@@ -6,14 +6,66 @@ export = React;
 export as namespace React;
 
 declare namespace React {
+  type Props = {[key: string]: any};
+  type State = Props;
+  type Context = Props;
+
   export class Component {
-    state: any;
-    props: any;
-    setState: Function;
-    constructor(props: any);
-    static propTypes: any;
-    static defaultProps: any;
+    context: Context;
+    props: Props;
+    state: State;
+    setState: (partialState: State, cb?: Function) => void;
+    constructor(props: Props, context?: Context);
   }
+
+  export class PureComponent extends Component {}
+
   function createElement(...args:any[]): any;
-  var PropTypes : any;
+
+  var Children: ReactChildren;
+  var PropTypes: ReactPropTypes;
+
```

Comment:
the below defs are copied from @types/react to support `React.Children` and `React.PropTypes` above